components/script/dom/htmlinputelement.rs

```diff
@@ -1704,6 +1703,25 @@ fn broadcast_radio_checked(broadcaster: &HTMLInputElement, group: Option<&Atom>)
     }
 }
 
+fn perform_radio_group_validation(elem: &HTMLInputElement, group: Option<&Atom>, can_gc: CanGc) {
+    for r in radio_group_iter(elem, group) {
+        r.validity_state()
+            .perform_validation_and_update(ValidationFlags::all(), can_gc);
+    }
+}
+
+fn perform_validation_conditionaly(elem: &HTMLInputElement, can_gc: CanGc) {
```

Comment:
Let's name this `update_related_validity_states`. Right now the name sounds like we may not update anything at all.